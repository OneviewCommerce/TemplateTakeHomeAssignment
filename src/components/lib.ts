// COMPONENTS EXPORTER
export { App } from "./app/app";

export { Nav } from "./nav/nav";
export { Table } from "./table/table";
export { Loading } from "./loading/loading";
export { Message } from "./message/message";
