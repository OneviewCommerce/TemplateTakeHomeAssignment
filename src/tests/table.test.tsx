import { renderWithRedux } from "./helper";
import { shallow } from "enzyme";
import { testUsers } from "./helper";
import { Table } from "../components/lib";

describe("Table Component", () => {
  it("renders with no errors", () => {
    let wrapper = shallow(<Table />);
    expect(wrapper.length).toBe(1);
  });

  it("renders a table with data in snapshot", () => {
    let wrapper = renderWithRedux(
      <Table
        tableData={testUsers}
        tableColumns={[
          {
            Header: "Name",
            accessor: "name",
          },
          {
            Header: "Email",
            accessor: "email",
          },
          {
            Header: "City",
            accessor: "address.city",
          },
          {
            Header: "Company",
            accessor: "company.name",
          },
        ]}
      />,
      { initialState: {} }
    );
    expect(wrapper).toMatchSnapshot();
  });
});
