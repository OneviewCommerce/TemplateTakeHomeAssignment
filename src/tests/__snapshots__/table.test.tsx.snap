// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component renders a table with data in snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table"
        role="table"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              colspan="1"
              role="columnheader"
            >
              Name
              <div
                class="search"
              >
                <input
                  placeholder="Search 2 records"
                  value=""
                />
              </div>
            </th>
            <th
              colspan="1"
              role="columnheader"
            >
              Email
              <div
                class="search"
              >
                <input
                  placeholder="Search 2 records"
                  value=""
                />
              </div>
            </th>
            <th
              colspan="1"
              role="columnheader"
            >
              City
              <div
                class="search"
              >
                <input
                  placeholder="Search 2 records"
                  value=""
                />
              </div>
            </th>
            <th
              colspan="1"
              role="columnheader"
            >
              Company
              <div
                class="search"
              >
                <input
                  placeholder="Search 2 records"
                  value=""
                />
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="row"
            role="row"
          >
            <td
              role="cell"
            >
              Hire me
            </td>
            <td
              role="cell"
            >
              user@email.com
            </td>
            <td
              role="cell"
            >
              city
            </td>
            <td
              role="cell"
            >
              company
            </td>
          </tr>
          <tr
            class="row"
            role="row"
          >
            <td
              role="cell"
            >
              User2
            </td>
            <td
              role="cell"
            >
              user2@email.com
            </td>
            <td
              role="cell"
            >
              city2
            </td>
            <td
              role="cell"
            >
              company2
            </td>
          </tr>
        </tbody>
      </table>
      <div>
        <button
          disabled=""
        >
          &lt;&lt;
        </button>
         
        <button
          disabled=""
        >
          &lt;
        </button>
         
        <button
          disabled=""
        >
          &gt;
        </button>
         
        <button
          disabled=""
        >
          &gt;&gt;
        </button>
         
        <span>
          Page
           
          <strong>
            1
             of 
            1
          </strong>
           
        </span>
        <select>
          <option
            value="2"
          >
            Show 2
          </option>
          <option
            value="3"
          >
            Show 3
          </option>
          <option
            value="5"
          >
            Show 5
          </option>
          <option
            value="7"
          >
            Show 7
          </option>
          <option
            value="10"
          >
            Show 10
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <table
      class="table"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
          >
            Name
            <div
              class="search"
            >
              <input
                placeholder="Search 2 records"
                value=""
              />
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Email
            <div
              class="search"
            >
              <input
                placeholder="Search 2 records"
                value=""
              />
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            City
            <div
              class="search"
            >
              <input
                placeholder="Search 2 records"
                value=""
              />
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Company
            <div
              class="search"
            >
              <input
                placeholder="Search 2 records"
                value=""
              />
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          class="row"
          role="row"
        >
          <td
            role="cell"
          >
            Hire me
          </td>
          <td
            role="cell"
          >
            user@email.com
          </td>
          <td
            role="cell"
          >
            city
          </td>
          <td
            role="cell"
          >
            company
          </td>
        </tr>
        <tr
          class="row"
          role="row"
        >
          <td
            role="cell"
          >
            User2
          </td>
          <td
            role="cell"
          >
            user2@email.com
          </td>
          <td
            role="cell"
          >
            city2
          </td>
          <td
            role="cell"
          >
            company2
          </td>
        </tr>
      </tbody>
    </table>
    <div>
      <button
        disabled=""
      >
        &lt;&lt;
      </button>
       
      <button
        disabled=""
      >
        &lt;
      </button>
       
      <button
        disabled=""
      >
        &gt;
      </button>
       
      <button
        disabled=""
      >
        &gt;&gt;
      </button>
       
      <span>
        Page
         
        <strong>
          1
           of 
          1
        </strong>
         
      </span>
      <select>
        <option
          value="2"
        >
          Show 2
        </option>
        <option
          value="3"
        >
          Show 3
        </option>
        <option
          value="5"
        >
          Show 5
        </option>
        <option
          value="7"
        >
          Show 7
        </option>
        <option
          value="10"
        >
          Show 10
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
